src/components/SubNavigation/SubNavigation.tsx

```diff
@@ -0,0 +1,179 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+
+import {ArrowShapeTurnUpRight, SquareListUl} from '@gravity-ui/icons';
+import {Button} from '@gravity-ui/uikit';
+import block from 'bem-cn-lite';
+
+import './SubNavigation.scss';
+
+const b = block('dc-subnavigation');
+
+export type ShareData = {
+    title: string | undefined;
+    url?: string;
+};
+
+export interface SubNavigationProps {
+    title: string | undefined;
+    hideMiniToc: boolean;
+    miniTocOpened: boolean;
+    toggleMiniTocOpen: () => void;
+    closeMiniToc: () => void;
+    // shareData: ShareData;
+}
+
+export const SubNavigation = ({
+    title,
+    hideMiniToc,
+    miniTocOpened,
+    // shareData
+    toggleMiniTocOpen,
+    closeMiniToc,
+}: SubNavigationProps) => {
+    const [visibility, setVisibility] = useState(true);
+    const [hiddingTimeout, setHiddingTimeout] = useState<number | undefined>(undefined);
+    const [lastScrollY, setLastScrollY] = useState(window.screenY);
+
+    const shareData = useMemo(() => {
+        return {
+            title,
+            // url: window.location.href // ?
+        };
+    }, [title]);
+
+    const clickOutsideMiniToc = useCallback(
+        (event: MouseEvent) => {
+            /*
+             * func "composedPath" returns an array in which the last two elements are "HTML" and "#document",
+             * which do not have the classList property, so they are subtracted before checking by slice()
+             */
+            const isOutside = !event
+                .composedPath()
+                .slice(0, -2)
+                .some((item) => {
+                    const el = item as HTMLElement;
+                    const classes = el.classList ?? [];
+
+                    return classes?.contains('dc-doc-layout__right');
+                });
+
+            if (isOutside) {
```

Comment:
Maybe it's better to add this for stopping further actions?
```
event.preventDefault();
event.stopPropagation();
```

Reply:
These functions broke the links in the mini-toc and broke the button that opens the mini-toc. I think, It is better not to use them